### ISSUE TITLE
Project still runs the old code after repository updated in pod 

### ISSUE THREAD

**Issue by davidear** [CocoaPods/CocoaPods#5369]
> ℹ Please fill out this template when filing an issue.
> All lines beginning with an ℹ symbol instruct you with
> what info we expect.  
> Please remove this line and all above before submitting.
- [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).
# Report
## What did you do?

 My project was using pod 0.39.0 for several months  
1. As recommended, I update my pod version 0.39.0 to 1.0.0.  
2. I has a third repository with version 0.5 to update to 0.6, so I use `pod update`  
3. I run the project   
## What did you expected to happen?

The project works with the new code of the third repository
## What happened instead?

Within the new pod 1.0.0 version:  
I found that, the project still use the old version 0.5 code of the third repository.  
In this situation, I check the repository's version, it's correct: 0.6  
I check the code in `Pods` directory, it's the correct code in 0.6 which I expected.    

And Finally, I return back pod version from 1.0.0 to 0.39.0, everything works fine.When I update the repository to 0.6, the project runs with the 0.6 code, when I degrade the repository to 0.5, the project runs with the 0.5 code, perfect.  
And I tried several repositories to confirm it, and I got the same result.   
## CocoaPods Environment

I found the problem in pod version 1.0.0, and it is fine in 0.39.0  
I am using Obj-C, and the podfile is:  
          source 'https://github.com/CocoaPods/Specs.git'
          platform :ios, '8.0'
          inhibit_all_warnings!
          .......
## Project that demonstrates the issue

Pick a repository somewhere and do the test is ok, I think.  
You need to run the project before the repository updated.

**Comment by orta** [comment]
This issue is hard to follow, can you write it out as a script or with an example project?

**Comment by neonichu** [comment]
It sounds like this is a duplicate of #5341 — as you're saying the code in the Pods directory is the expected one, it doesn't sound like a caching problem.

**Comment by valleyzhx** [comment]
I meet this same problem
